app/models/user.rb

```diff
@@ -95,6 +95,10 @@ def to_s
     name || login
   end
 
+  def self.find_by_login(val)
```

Comment:
*I know that I am the new guy.* I would suggest to place this **class-level** (i.e., `self`) method together with the other **class-level** methods above.